### PR TITLE
crowbar_register: Set the target platform of the node we're registering

### PR DESCRIPTION
**Pull request by vuntz**
Otherwise, it'll default to the default of the admin server, which might
not be correct (ie, not SLE12).

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
so this sets the target platform in the UI -> node view?

**Comment by vuntz** [comment]
@MaximilianMeister this sets the attribute that the UI is using. But other bits are using this attribute (some cookbooks for instance). That's why it's important to get it right.

**Comment by MaximilianMeister** [comment]
good to know.
+1

**Comment by tboerger** [comment]
LGTM

File: chef/cookbooks/provisioner/templates/suse/crowbar_register.erb [hunk 1]
```diff
@@ -285,8 +285,13 @@ curl -L -o /etc/chef/validation.pem \
     --connect-timeout 60 -s \
     "$HTTP_SERVER/validation.pem"
 
+TMP_ATTRIBUTES=$(mktemp --suffix .json)
+
+# Make sure that we have the right target platform
+echo "{ \"target_platform\": \"$CROWBAR_OS\" }" > "$TMP_ATTRIBUTES"
```
Comment (dirkmueller) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: does that add it as normal attributes or as default attribute ? in mkcloud we're setting normal.target_platform via knife node edit

Comment (vuntz) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: Hrm, I don't know, but I would expect normal attribute. Would need to try again to be sure.

Comment (MaximilianMeister) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: > does that add it as normal attributes or as default attribute ? in mkcloud we're setting normal.target_platform via knife node edit

its added as "normal", i just checked

**Pull request merged**